File: code/C/第1章 数据结构与算法/1.3 基础算法/reverse.c
#include <stdio.h>
#include <string.h>

void reverse(char *str) {
    int i = 0;
    int j = strlen(str) - 1;
    while(i < j) {
        char temp = str[i];
        str[i] = str[j];
        str[j] = temp;
        i++;
        j--;
    }
}

int main() {
    char str[] = "hello";
    reverse(str);
    printf("%s\n", str);
    return 0;
}

File: code/Python/第1章 数据结构与算法/1.3 基础算法/reverse.py
def reverse(s):
    s = list(s)
    i = 0
    j = len(s) - 1
    while i < j:
        s[i], s[j] = s[j], s[i]
        i += 1
        j -= 1
    return "".join(s)

def main():
    str = "hello"
    print(reverse(str))

if __name__ == "__main__":
    main()